jina/types/document/__init__.py

```diff
@@ -954,16 +997,122 @@ def score(
+        index: int,
+        value: Union[jina_pb2.NamedScoreProto, NamedScore, float, np.generic],
+    ):
+        """Set an evaluation of the document.
+
+        You can assign a scalar variable directly.
+
+        :param index: the index of the evaluations list to set the value in
+        :param value: the value to set the evaluation of the Document from
+        """
         if isinstance(value, jina_pb2.NamedScoreProto):
-            self._pb_body.score.CopyFrom(value)
+            self._pb_body.evaluations[index].CopyFrom(value)
         elif isinstance(value, NamedScore):
-            self._pb_body.score.CopyFrom(value._pb_body)
+            self._pb_body.evaluations[index].CopyFrom(value._pb_body)
         elif isinstance(value, (float, int)):
-            self._pb_body.score.value = value
+            self._pb_body.evaluations[index].value = value
         elif isinstance(value, np.generic):
-            self._pb_body.score.value = value.item()
+            self._pb_body.evaluations[index].value = value.item()
         else:
-            raise TypeError(f'score is in unsupported type {typename(value)}')
+            raise TypeError(f'evaluation is in unsupported type {typename(value)}')
+
+    @property
+    def evaluation(self):
+        """Return the first evaluation of the document evaluation.
+            This interface is intended to be used in simple cases where a single `evaluation` metric is used.
+
+        :return: the evaluation attached to this document as `:class:NamedScore`
+        """
+        if len(self._pb_body.evaluations) > 0:
+            return NamedScore(self._pb_body.evaluations[0])
+
+    @evaluation.setter
+    def evaluation(
+        self, value: Union[jina_pb2.NamedScoreProto, NamedScore, float, np.generic]
+    ):
+        """Set the evaluation of the document.
+
+        You can assign a scala variable directly.
+
+        :param value: the value to set the evaluation of the Document from
+        """
+        if len(self._pb_body.evaluations) == 0:
+            self._pb_body.evaluations.add()
+        self._set_evaluation(0, value)
+        if not self.evaluation.op_name:
+            self.evaluation.op_name = _default_evaluation_name
+
+    def set_evaluation(
+        self,
+        name: str,
+        value: Union[jina_pb2.NamedScoreProto, NamedScore, float, np.generic],
+    ):
+        """Set the evaluation of the document for a given evaluation name.
+
+        You can assign a scalar variable directly.
+
+        :param name: the operation name of the evaluation to be added
+        :param value: the value to set the evaluation of the Document from
+        """
+        idx = self._evaluations_offset_map.get(name, len(self.evaluations))
+        if idx == len(self.evaluations):
+            self._pb_body.evaluations.add()
+        self._set_evaluation(idx, value)
+        self._pb_body.evaluations[idx].op_name = name
+
+    def get_evaluation(self, name: str) -> Optional['NamedScore']:
```

Comment:
why not set as property?

Reply:
There is a `property` to access the `first` element, which is kind of syntactic sugar. Let me explain better in the `description`